gluonnlp/model/beam_search.py

```diff
@@ -92,30 +92,56 @@ def _expand_to_beam_size(data, beam_size, batch_size):
         Each NDArray should have shape (batch_size * beam_size, ...)
     """
     if isinstance(data, list):
-        return [_expand_to_beam_size(ele, beam_size, batch_size) for ele in data]
+        assert not state_info or isinstance(state_info, list), \
```

Comment:
I think we can move this part outside if else by using 
`assert not state_info or isinstance(state_info, type(data))`